### PR TITLE
fix(titus): Don't copy migrationPolicy from source during job creation

### PR DESCRIPTION
**Pull request by aravindmd**
- We use to copy the `migrationPolicy` from source even when it was not specified in the payload and `jobDisruptionBudget` was set as part of the payload, this fix it to ignore the source `migrationPolicy` when `jobDisruptioBuget` is set and copy only when its not specified.

### CODE REVIEW

File: clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/deploy/handlers/TitusDeployHandler.groovy [hunk 1]
```diff
@@ -172,7 +173,9 @@ class TitusDeployHandler implements DeployHandler<TitusDeployDescription> {
         if (description.inService == null) {
           description.inService = sourceJob.inService
         }
-        description.migrationPolicy = description.migrationPolicy ?: sourceJob.migrationPolicy
+        if(description.disruptionBudget == null) {
```
Comment (ajordens) [inline clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/deploy/handlers/TitusDeployHandler.groovy]: space before the `(` and perhaps an inline comment saying that the `migrationPolicy` should only be used when the `disruptionBudget` has not been specified.

Comment (ajordens) [reply clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/deploy/handlers/TitusDeployHandler.groovy]: what if both are specified?

Comment (ajordens) [reply clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/deploy/handlers/TitusDeployHandler.groovy]: Should the `disruptionBudget` be propagated from the `sourceJob`? (is that a thing?)

File: clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/deploy/handlers/TitusDeployHandler.groovy [hunk 2]
```diff
@@ -16,6 +16,7 @@
 
 package com.netflix.spinnaker.clouddriver.titus.deploy.handlers
 
+import com.fasterxml.jackson.databind.ObjectMapper
```
Comment (ajordens) [inline clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/deploy/handlers/TitusDeployHandler.groovy]: where did this import come from?

File: clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/deploy/handlers/TitusDeployHandler.groovy (hunk 1 above)
Comment (aravindmd) [reply clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/deploy/handlers/TitusDeployHandler.groovy]: good question , let me circle back on this with titus

Comment (aravindmd) [reply clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/deploy/handlers/TitusDeployHandler.groovy]: i think that will need some more work and most likely changes to the caching agents to, i will take a look at this next in a separate PR , this should fix the immediate issue i think.

File: clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/deploy/handlers/TitusDeployHandler.groovy (hunk 2 above)
Comment (aravindmd) [reply clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/deploy/handlers/TitusDeployHandler.groovy]: removed , i was using this for some local logging test

**Comment by ajordens** [comment]
LGTM

**Pull request merged**